include/olm/session.hh

```diff
@@ -51,15 +51,16 @@ struct OLM_EXPORT Session {
     /** Start a new outbound session. Returns std::size_t(-1) on failure. Assumes
      * prekey_signature is a byte array of ED25519_SIGNATURE_LENGTH characters. On
      * failure last_error will be set with an error code. The last_error will be
-     * NOT_ENOUGH_RANDOM if the number of random bytes was too small. */
+     * NOT_ENOUGH_RANDOM if the number of random bytes was too small.
+     * When one_time_key is nullptr uses prekey to perform three steps of DH. */
     std::size_t new_outbound_session(
         Account const & local_account,
         _olm_curve25519_public_key const & identity_key,
         _olm_ed25519_public_key const & signing_key,
         _olm_curve25519_public_key const & pre_key,
         std::uint8_t const * prekey_signature,
-        _olm_curve25519_public_key const & one_time_key,
-        std::uint8_t const * random, std::size_t random_length
+        std::uint8_t const * random, std::size_t random_length,
+        _olm_curve25519_public_key const * one_time_key = nullptr
```

Comment:
`.hh` is just a convention to differentiate between C/C++ headers (same as `.hpp`) - this is more about how this file is compiled/linked.
I wasn't diving into Makefile but since it has some only C++ features (namespace, `std::`) I assume we're safe to use other C++ features.